pygame/math.py

```diff
@@ -0,0 +1,1575 @@
+# pygame_cffi - a cffi implementation of the pygame library
+# Copyright (C) 2016  Anton Joubert
+# Copyright (C) 2016  Simon Cross
+# Copyright (C) 2016  Neil Muller
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public
+# License along with this library; if not, write to the Free
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+# MA  02110-1301  USA
+
+""" Math module """
+
+from __future__ import absolute_import, division
+import math
+from numbers import Number
+
+from pygame.compat import string_types, pow_compat
+
+VECTOR_EPSILON = 1e-6  # For equality tests
+
```

Comment:
I understand that you want to hide some of these details, but not sure how to use `__all__` for this.  If I add something like `__all__ = ['enable_swizzling', 'disable_swizzling', 'Vector2', 'Vector3']` to math.py, then the dir() and ipython autocomplete output is still the same.  It only makes a difference if I do `from pygame.math import *`.  Adding `__all__` to the top level `__init__.py` file would be hard, since I'd have to list every single function in the package.

How closely do we need to match pygame? 
E.g. It provides: 
```
In [15]: dir(pygame.math)
Out[15]: 
['Vector2',
 'Vector3',
 'VectorElementwiseProxy',
 'VectorIterator',
 '_PYGAME_C_API',
 '__doc__',
 '__file__',
 '__name__',
 '__package__',
 'disable_swizzling',
 'enable_swizzling']
```

So far, this module provides:
```
In [2]: dir(pygame.math)
Out[2]: 
['ElementwiseVector2Proxy',
 'ElementwiseVector3Proxy',
 'ElementwiseVectorProxyBase',
 'Number',
 'VECTOR_EPSILON',
 'Vector2',
 'Vector3',
 '__builtins__',
 '__doc__',
 '__file__',
 '__name__',
 '__package__',
 '_rotate_2d',
 'absolute_import',
 'disable_swizzling',
 'division',
 'enable_swizzling',
 'math',
 'pow_compat',
 'string_types']
```

I prefer @hodgestar's suggestion to use the `_` prefix to imply private use.  Is the same required for imports, e.g. `from numbers import Number` -> `from numbers import Number as _Number`, so that they don't show up so easily?